src/format/KdbxXmlReader.cpp

```diff
@@ -367,6 +367,10 @@ void KdbxXmlReader::parseIcon()
     }
 
     if (uuidSet && iconSet) {
+        // Check for duplicate UUID (corruption)
+        if (m_meta->containsCustomIcon(uuid)) {
+            uuid = QUuid::createUuid();
```

Comment:
Wow, this should be astronomically unlikely with a **U**UID unless someone messed with their database.

Reply:
That's exactly what happens! Or some other program jacked it up.